File: tests/loop/slides/loop_slides.cpp
#include <optitrust.h>
#include <stdio.h>

int main() {
  for (int i = 0; i < 15; i += 3) {
    for (int j = 0; j < 15; j += 3) {
      printf("%d, %d\n", i, j);
    }
  }

  return 0;
}

File: tests/loop/slides/loop_slides.ml
open Optitrust
open Target
open Prelude

let _ = Run.script_cpp (fun _ ->
  !! Loop.slides ~size_steps:[
    Some (trm_int 2, trm_int 1);
    Some (trm_int 3, trm_int 2);
  ] [cFor "i"];
)
